### PR TITLE
Added build.yml to run tests on pull requests

### PR DESCRIPTION
**Pull request by ayushsarode**
Kindly review my pull request. It addresses issue #64. Your feedback is greatly appreciated. Thank you!

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Build and Test
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-lates
```
Comment (rmgrimm) [inline .github/workflows/build.yml:10]: Looks like a typo here. Should be `ubuntu-latest`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
```
Comment (rmgrimm) [inline .github/workflows/build.yml:14]: Please use the latest version of `actions/checkout`, ie v4

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Build and Test
```
Comment (rmgrimm) [inline .github/workflows/build.yml:1]: How about a name of `Run Tests on Pull Requests`

**Comment by rmgrimm** [comment]
Thanks for the pull request! Can you describe `includes essential updates` in more detail? What updates does it include beyond the new workflow?

**Review by rmgrimm — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-lates
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
```
[the remark this answers is not in the document]
Comment (ayushsarode) [reply .github/workflows/build.yml:19]: Do I set it as ```distribution: 'adopt'```?

[the remark this answers is not in the document]
Comment (rmgrimm) [reply .github/workflows/build.yml]: No, use a value of `temurin`, not `adopt`. Per the [note on actions/setup-java's distributions list](https://github.com/actions/setup-java#supported-distributions), the `adopt` value is discontinued and replaced by `temurin`.

[the remark this answers is not in the document]
Comment (ayushsarode) [reply .github/workflows/build.yml]: Sure, Thanks!

**Review by rmgrimm — APPROVED** [review]

**Comment by rmgrimm** [comment]
Thanks for the pull request!

**Comment by ayushsarode** [comment]
Thank you @rmgrimm for helping! Your support was crucial, and I appreciate your time and expertise. Grateful.

**Comment by ayushsarode** [comment]
Also, Could you please add "hacktoberfest" topic to your repository?

**Comment by rmgrimm** [comment]
> Also, Could you please add "hacktoberfest" topic to your repository?

It is already on the repository

**Pull request merged**